client/app/components/navbar/navbar-notification.service.js

```diff
@@ -0,0 +1,109 @@
+class NavbarNotificationService {
+    constructor ($interval, $q, $translate, Alerter, constants, OvhApiNotificationAapi, translator, UNIVERSE) {
+        this.$interval = $interval;
+        this.$q = $q;
+        this.$translate = $translate;
+        this.alerter = Alerter;
+        this.constants = constants;
+        this.OvhApiNotificationAapi = OvhApiNotificationAapi;
+        this.translator = translator;
+        this.UNIVERSE = UNIVERSE;
+
+        this.NOTIFICATION_REFRESH_TIME = 60000;
+    }
+
+    getMessages () {
+        return this.$translate.refresh().then(() => this.OvhApiNotificationAapi.query({
+            lang: this.$translate.preferredLanguage(),
+            target: this.constants.target,
+            universe: this.UNIVERSE
+        }).$promise);
+    }
+
+    getSubLinks () {
+        return this.getMessages().then((messages) => messages.map((message) => this.convertSubLink(message))).catch(() => [{
+            id: "0",
+            date: moment(),
+            time: this._formatTime(moment()),
+            level: "error",
+            subject: this.translator.tr("common_navbar_notification_error_subject"),
+            description: this.translator.tr("common_navbar_notification_error")
+        }]);
+    }
+
+    /* eslint-disable class-methods-use-this */
+    _formatTime (dateTime) {
+        return moment(dateTime).fromNow();
+    }
+    /* eslint-enable class-methods-use-this */
+
+    _toggleSublinkAction (toUpdate, linkClicked) {
+        if (toUpdate.isActive && !toUpdate.updating) {
+            toUpdate.updating = true;
+            this.OvhApiNotificationAapi.post({ completed: [toUpdate.id] }).$promise.then(() => {
+                toUpdate.isActive = false;
+                toUpdate.acknowledged = true;
+            }).finally(() => { toUpdate.updating = false; });
```

Comment:
No catching ?

Reply:
The feedback I give to the client is good, if the notification can't be updated it's not shown as updated. I don't think mark a notif as read is critical and need to display a real feedback when it fails.